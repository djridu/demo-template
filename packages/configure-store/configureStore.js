/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();

export default function configureStore(initialState = {}, reducers = {}, middlewares = []) {
    const middlewaresToApply = [sagaMiddleware, ...middlewares];
    const enhancers = [applyMiddleware(...middlewaresToApply)];

    const composeEnhancers =
        '__DEV__' && typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
            ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
            : compose;

    const store = createStore(
        combineReducers({ ...reducers }),
        initialState,
        composeEnhancers(...enhancers),
    );

    store.runSaga = sagaMiddleware.run;
    store.injectedReducers = reducers;
    store.injectedSagas = {};

    if (module.hot) {
        module.hot.accept(() => {
            store.replaceReducer(combineReducers({ ...store.injectedReducers }));
        });
    }

    return store;
}
