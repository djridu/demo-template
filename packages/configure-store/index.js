// @flow

export { default as configureStore } from './configureStore';
