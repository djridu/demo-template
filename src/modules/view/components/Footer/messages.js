/*
 * Footer Messages
 */
import { defineMessages } from 'react-intl';

export const scope = 'components.Footer';

export default defineMessages({
    licenseMessage: {
        id: `${scope}.license.message`,
        defaultMessage: 'qwe',
    },
    authorMessage: {
        id: `${scope}.author.message`,
        defaultMessage: 'qwe',
    },
});
