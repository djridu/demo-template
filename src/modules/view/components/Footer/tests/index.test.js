import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider, ReactReduxContext } from 'react-redux';

import { configureStore } from 'configure-store';
import languageProviderReducer from 'containers/LanguageProvider/reducer';
import Footer from '../index';

describe('<Footer />', () => {
    let store;

    beforeAll(() => {
        const initialState = {
            language: {
                locale: 'de',
            },
        };
        store = configureStore(initialState, { language: languageProviderReducer }, []);
    });

    it('should render and match the snapshot', () => {
        const renderedComponent = renderer
            .create(
                <Provider store={store} context={ReactReduxContext}>
                    <IntlProvider locale="en">
                        <Footer />
                    </IntlProvider>
                </Provider>,
            )
            .toJSON();

        expect(renderedComponent).toMatchSnapshot();
    });
});
