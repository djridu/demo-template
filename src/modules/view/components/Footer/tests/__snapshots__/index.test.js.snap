// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render and match the snapshot 1`] = `
<footer
  className="Wrapper-zrtjx0-0 bJElKi"
>
  <section>
    <div
      className="Wrapper-sc-1xsw5mc-0 JDLrM"
    >
      <select
        className="Select-sc-6vw2nw-0 gcyFDJ"
        onChange={[Function]}
        value="de"
      >
        <option
          value="en"
        >
          en
        </option>
        <option
          value="de"
        >
          de
        </option>
      </select>
    </div>
  </section>
</footer>
`;
